### ISSUE TITLE
after fix "no space to left", container restart policy become invalid

### ISSUE THREAD

**Issue by dragon9783** [moby/moby#38356]
**Description**

**Steps to reproduce the issue:**
1. start kafka container with restart policy `always`
2. write data into kafka via api or sh
3. `df -h` host space become 100%

**Describe the results you received:**
kafka container become restarting status 

**Describe the results you expected:**
kafka container become Running status

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `docker version`:**

```
Client:
 Version:      18.03.1-ce
 API version:  1.37
 Go version:   go1.9.5
 Git commit:   9ee9f40
 Built:        Thu Apr 26 07:17:20 2018
 OS/Arch:      linux/amd64
 Experimental: false
 Orchestrator: swarm

Server:
 Engine:
  Version:      18.03.1-ce
  API version:  1.37 (minimum version 1.12)
  Go version:   go1.9.5
  Git commit:   9ee9f40
  Built:        Thu Apr 26 07:15:30 2018
  OS/Arch:      linux/amd64
  Experimental: false
```

**Output of `docker info`:**

```
Containers: 34
 Running: 33
 Paused: 0
 Stopped: 1
Images: 92
Server Version: 18.03.1-ce
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 773c489c9c1b21a6d78b5c538cd395416ec50f88
runc version: 4fc53a81fb7c994640722ac585fa9ca548971871
init version: 949e6fa
Security Options:
 apparmor
 seccomp
  Profile: default
Kernel Version: 4.4.0-131-generic
Operating System: Ubuntu 16.04.5 LTS
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 31.42GiB
Name: ubuntu
ID: GF4L:OGXY:IZUW:5VJN:CVCA:KM3L:N5D5:55OI:B5YM:ACPD:QIQB:YBAZ
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Labels:
Experimental: false
Insecure Registries:
 registry.supos.ai
 127.0.0.0/8
Live Restore Enabled: false
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by dragon9783** [comment]
**container config.v2.json:**
"State":{"Running":true,"Paused":false,"Restarting":true,"OOMKilled":false,"RemovalInProgress":false,"Dead":false,"Pid":0,"ExitCode":1,"Error":"","StartedAt":"2018-12-13T01:04:43.645177367Z","FinishedAt":"2018-12-13T06:58:51.449215527Z","Health":null}

**docker inspect container:**
"State": {
            "Status": "exited",
            "Running": false,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 0,
            "ExitCode": 1,
            "Error": "write /var/lib/docker/containerd/daemon/io.containerd.metadata.v1.bolt/meta.db: no space left on device: unknown",
            "StartedAt": "2018-12-13T01:04:43.645177367Z",
            "FinishedAt": "2018-12-13T06:58:51.377943957Z"
        }
restarting status is different

**Comment by thaJeztah** [comment]
I'm not sure there's much that can be done in this situation; the host is out of disk space, thus writing changes to the container state fails (or storage gets corrupted). The docker daemon in your situation apparently just has enough resources to keep running, detects the container fails to run successfully, thus attempts to restart the container (which fails, due to the host being our of disk space).

Tl;DR; if your host is out of disk space, bad things can happen, and all bets are off

**Comment by dragon9783** [comment]
Thanks to response soonly, We increase our monitoring capability to host, alert when host almost being out of disk space.
